### PR TITLE
Update WebIDL.py

### PR DESCRIPTION
**Pull request by wusyong**
We found some codegen error when trying to add [queuing strategy](https://streams.spec.whatwg.org/#qs-api), but firefox has exact webidl and works fine.
It's time to update servo's webidl.py to fix this issue.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [x] There are tests for these changes. WPT suite should pass.

### CODE REVIEW

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -3980,7 +3980,10 @@ def definition_body(self):
 let args = CallArgs::from_vp(vp, argc);
 let global = GlobalScope::from_object(args.callee());
 """
-        if len(self.exposureSet) == 1:
+
+        if len(self.exposureSet) == 2 and "Window" in self.exposureSet:
+            preamble += "let global = DomRoot::downcast::<dom::types::Window>(global).unwrap();\n"
```
[the remark this answers is not in the document]
Comment (wusyong) [reply components/script/dom/bindings/codegen/CodegenRust.py:3985]: This and the below checks are for exposureSet, which contains the exact `["DissimimalrOriginWindow" "Window"]`.
Before this update, they are just `["Window"]`. It seems `DissimimalrOriginWindow` is for internal usage. So in this case it will always be "Window". If you want, I can add more checks to make sure the set is exactly `["DissimimalrOriginWindow" "Window"]`

File: components/script/dom/webidls/DissimilarOriginWindow.webidl [hunk 2]
```diff
@@ -13,7 +13,7 @@
 // way to enforce security policy.
 
 // https://html.spec.whatwg.org/multipage/#window
-[Global, Exposed=(Window,DissimilarOriginWindow), LegacyNoInterfaceObject]
+[Global=(Window,DissimilarOriginWindow), Exposed=(Window,DissimilarOriginWindow), LegacyNoInterfaceObject]
```
Comment (jdm) [inline components/script/dom/webidls/DissimilarOriginWindow.webidl:16]: Why is this change necessary?

Comment (jdm) [reply components/script/dom/webidls/DissimilarOriginWindow.webidl:16]: More specifically, why not `Global=DissimilarOriginWindow`?

**Comment by jdm** [comment]
The following diff builds for me:
```
diff --git a/components/script/dom/bindings/codegen/CodegenRust.py b/components/script/dom/bindings/codegen/CodegenRust.py
index 5bb0a8aaec..e3ff49930f 100644
--- a/components/script/dom/bindings/codegen/CodegenRust.py
+++ b/components/script/dom/bindings/codegen/CodegenRust.py
@@ -3981,12 +3981,7 @@ let args = CallArgs::from_vp(vp, argc);
 let global = GlobalScope::from_object(args.callee());
 """

-        # Exposure set {'DissimilarOriginWindow', 'Window'} should downcast to Window.
-        # DissimilarOriginWindow is not for external usage.
-        if len(self.exposureSet) == 2 and "Window" in self.exposureSet and "DissimilarOriginWindow" in self.exposureSet:
-            print(self.exposureSet)
-            preamble += "let global = DomRoot::downcast::<dom::types::Window>(global).unwrap();\n"
-        elif len(self.exposureSet) == 1:
+        if len(self.exposureSet) == 1:
             preamble += ("let global = DomRoot::downcast::<dom::types::%s>(global).unwrap();\n" %
                          list(self.exposureSet)[0])
         return CGList([CGGeneric(preamble), self.generate_code()])
@@ -6136,11 +6131,7 @@ let args = CallArgs::from_vp(vp, argc);
 let global = GlobalScope::from_object(JS_CALLEE(*cx, vp).to_object());
 """

-        # Exposure set {'DissimilarOriginWindow', 'Window'} should downcast to Window.
-        # DissimilarOriginWindow is not for external usage.
-        if len(self.exposureSet) == 2 and "Window" in self.exposureSet and "DissimilarOriginWindow" in self.exposureSet:
-            preamble += "let global = DomRoot::downcast::<dom::types::Window>(global).unwrap();\n"
-        elif len(self.exposureSet) == 1:
+        if len(self.exposureSet) == 1:
             preamble += """\
 let global = DomRoot::downcast::<dom::types::%s>(global).unwrap();
 """ % list(self.exposureSet)[0]
diff --git a/components/script/dom/webidls/DissimilarOriginWindow.webidl b/components/script/dom/webidls/DissimilarOriginWindow.webidl
index 2e55d33f00..bce8bddb57 100644
--- a/components/script/dom/webidls/DissimilarOriginWindow.webidl
+++ b/components/script/dom/webidls/DissimilarOriginWindow.webidl
@@ -13,7 +13,7 @@
 // way to enforce security policy.

 // https://html.spec.whatwg.org/multipage/#window
-[Global=(Window,DissimilarOriginWindow), Exposed=(Window,DissimilarOriginWindow), LegacyNoInterfaceObject]
+[Global=DissimilarOriginWindow, Exposed=(Window,DissimilarOriginWindow), LegacyNoInterfaceObject]
 interface DissimilarOriginWindow : GlobalScope {
   [LegacyUnforgeable] readonly attribute WindowProxy window;
   [BinaryName="Self_", Replaceable] readonly attribute WindowProxy self;
```

File: components/script/dom/webidls/DissimilarOriginWindow.webidl (hunk 2 above)
Comment (wusyong) [reply components/script/dom/webidls/DissimilarOriginWindow.webidl:16]: Ah thanks. The new webidl.py raised this error so I added the identifier list:
```
    File "/Users/wuwayne/Desktop/servo/third_party/WebIDL/WebIDL.py", line 2059, in addExtendedAttributes
      raise WebIDLError(
  WebIDL.WebIDLError: error: [Global] must either take an identifier or take an identifier list, /Users/wuwayne/Desktop/servo/components/script/dom/bindings/codegen/../../webidls/DissimilarOriginWindow.webidl line 16:1
  [Global, Exposed=(Window,DissimilarOriginWindow), LegacyNoInterfaceObject]
   ^
 ```

Your suggestion is better and don't require fixing CodegenRust.py. I'll update it.

**Review by jdm — APPROVED** [review]
